src/storage/UpdateEdgeProcessor.cpp

```diff
@@ -364,10 +364,22 @@ void UpdateEdgeProcessor::process(const cpp2::UpdateEdgeRequest& req) {
             return std::string("");
         },
         [this, partId, edgeKey, req] (kvstore::ResultCode code) {
-            this->pushResultCode(this->to(code), partId);
+            cpp2::ResultCode thriftResult;
+            thriftResult.set_code(to(code));
+            thriftResult.set_part_id(partId);
+            if (code == kvstore::ResultCode::ERR_LEADER_CHANGED) {
+                nebula::cpp2::HostAddr leader;
+                auto addrRet = kvstore_->partLeader(this->spaceId_, partId);
+                CHECK(ok(addrRet));
```

Comment:
Here should return failure.

Reply:
> Here should return failure.

Why?

Reply:
We can't crash the process at will, we must handle the errors that can be handled.

PS: What I mean here is that you need to determine whether addrRet is legal.

Reply:
> We can't crash the process at will, we must handle the errors that can be handled.
> 
> PS: What I mean here is that you need to determine whether addrRet is legal.

You are right